airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```diff
@@ -147,7 +148,7 @@ protected AirbyteMessage computeNext() {
       return stateMessage;
     }
 
-    if (changeEventIterator.hasNext()) {
+    while (changeEventIterator.hasNext()) {
```

Comment:
this feels a pretty significant change - isn't the enclosing method getting called via the iterator pattern so that this method does not have to a `while` loop on the `changeEventIterator`?

Reply:
If we don't return an airbyte message, the caller iterator is going to fail.
I will verify, But I think we need to keep trying to get the next until we're able to return a record, or wrap up

Reply:
This feels like another example where we need to have different implementations:  one for relational databases and one for Mongo due to the behavior differences.  I agree with @prateekmukhedkar that we probably don't want to change how the iterator behaves for all CDC connectors -- just the Mongo one, unless we believe that this issue exists in all connector variants.

Reply:
Perhaps the cleanest way to tackle this right now is to catch this mongo specific error in a mongo connector level iterator